J-Stage.js

```diff
@@ -94,29 +101,35 @@ function scrape(doc, url) {
 		translator.setHandler("itemDone", function (obj, item) {
 			if (abs) item.abstractNote = abs;
 			if (tags) item.tags = tags;
-			for (i in item.creators) {
+			for (var i in item.creators) {
 				if (item.creators[i].lastName && item.creators[i].lastName == item.creators[i].lastName.toUpperCase()) {
-					item.creators[i].lastName = Zotero.Utilities.capitalizeTitle(item.creators[i].lastName.toLowerCase(), true);
+					item.creators[i].lastName = ZU.capitalizeTitle(item.creators[i].lastName.toLowerCase(), true);
 				}
 				if (item.creators[i].firstName && item.creators[i].firstName == item.creators[i].firstName.toUpperCase()) {
-					item.creators[i].firstName = Zotero.Utilities.capitalizeTitle(item.creators[i].firstName.toLowerCase(), true);
+					item.creators[i].firstName = ZU.capitalizeTitle(item.creators[i].firstName.toLowerCase(), true);
 				}
 			}
 			if (item.title == item.title.toUpperCase()) {
-				item.title = Zotero.Utilities.capitalizeTitle(item.title.toLowerCase(), true);
+				item.title = ZU.capitalizeTitle(item.title.toLowerCase(), true);
 			}
 			if (item.publicationTitle == item.publicationTitle.toUpperCase()) {
-				item.publicationTitle = Zotero.Utilities.capitalizeTitle(item.publicationTitle.toLowerCase(), true);
+				item.publicationTitle = ZU.capitalizeTitle(item.publicationTitle.toLowerCase(), true);
 			}
-			item.attachments = [{
-				url: item.url,
+			item.attachments.push({
+				url: url,
 				title: "Jstage - Snapshot",
```

Comment:
Let's use J-Stage here and l. 129 for the PDF, in line with the library catalog and translator label.